provider/pkg/gen/schema.go

```diff
@@ -294,6 +298,44 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 				pkg.Types[tok] = pschema.ComplexTypeSpec{
 					ObjectTypeSpec: objectSpec,
 				}
+
+				// Add "Patch" types corresponding to all non "List" resource kinds.
+				var patchSpec pschema.ObjectTypeSpec
+				if len(patchTok) > 0 {
+					patchSpec = pschema.ObjectTypeSpec{
+						Description: kind.Comment() + kind.PulumiComment(),
+						Type:        "object",
+						Properties:  map[string]pschema.PropertySpec{},
+						Language:    map[string]pschema.RawMessage{},
+					}
+
+					var propNames []string
+					for _, p := range kind.Properties() {
+						patchSpec.Properties[p.name] = genPropertySpec(p, kind.apiVersion, kind.kind+"Patch")
```

Comment:
Somewhat subtle that no required properties are being set (by design for patch resources). Might be worth calling out with a comment. We might want to add name and/or namespace (for namespaced objects back though)?